rosetta/api.go

```diff
@@ -18,7 +19,78 @@ import (
 var log = logger.GetOrCreate("rosetta")
 
 // CreateServer creates a HTTP server
-func CreateServer(elrondFacade api.ElrondProxyHandler, generalConfig *config.Config, port int) (*http.Server, error) {
+func CreateServer(
+	elrondFacade api.ElrondProxyHandler,
+	generalConfig *config.Config,
+	port int,
+	isOffline bool,
+) (*http.Server, error) {
+	if !isOffline {
+		return createOnlineServer(elrondFacade, generalConfig, port)
+	}
+
+	cfg := configuration.LoadOfflineConfig(generalConfig)
```

Comment:
This method and createOnlineServer can be refactored to remove duplicated code.
some of the lines 32-> 53 can be extracted into a CreateProvider method that receives either an offline or an online service
some of the lines 59 -> 82 can be extracted into a CreateHttpServer method

Reply:
refactored.